pocs/core.py

```diff
@@ -403,6 +406,60 @@ def has_free_space(self, required_space=0.25 * u.gigabyte):
         free_space = get_free_space()
         return free_space.value >= required_space.to(u.gigabyte).value
 
+    def has_ac_power(self, stale=90):
+        """Check for system AC power.
+
+        Power readings are done by the arduino and are placed in the metadata
+        database. This method looks for entries saved under the `power.main`
+        key in the `current` collection. The method will also return False if
+        the record is older than `stale` seconds.
+
+        Args:
+            stale (int, optional): Number of seconds before record is stale,
+                defaults to 90 seconds.
+
+        Returns:
+            bool: True if system AC power is present.
+        """
+        # Always assume False
+        self.logger.debug("Checking for AC power")
+        has_power = False
+
+        # TODO(wtgee): figure out if we really want to simulate no power
+        # Check if we are using power simulator
+        with suppress(KeyError):
+            if 'power' in self.config['simulator']:
+                self.logger.debug("AC power simulator always safe")
+                return True
+
+        # Get current power readings from database
+        try:
+            record = self.db.get_current('power')
+            has_power = bool(record['data'].get('main', False))
+
+            timestamp = record['date'].replace(tzinfo=None)  # current_time is timezone naive
+            age = (current_time().datetime - timestamp).total_seconds()
+
+            self.logger.debug("Power Safety: {} [{:.0f} sec old - {:%Y-%m-%d %H:%M:%S}]",
```

Comment:
That format is worth of an f-string! ;-)

Reply:
Actually this is a place where it doesn't help because then the string is longer than the line-length.

Reply:
Sigh